ex1/exercise 1.1 tests

```diff
@@ -0,0 +1,145 @@
+            [emp, emp, emp],
+        ])
+        assert not board.is_move_legal(0, 2)
+        board.play_move(0, 2, X)
+        assert not board.is_winning(X)
+
+    def test_out_of_bound_is_illegal_move(self):
+        board = XOBoard.empty_board()
+        assert board.is_move_legal(0, 0)
+        assert board.is_move_legal(2, 2)
+        assert not board.is_move_legal(-1, 0)
+        assert not board.is_move_legal(3, 0)
+        assert not board.is_move_legal(0, 3)
+
+    def test_changing_value_only_one_value_change(self):
+        board = XOBoard.empty_board()
+
+        board.play_move(0, 0, X)
+        assert board.is_move_legal(1, 0)
+
+
+class TestGame(unittest.TestCase):
+    def test_change_player(self):
+        game = XOGame([(0, 0), (1, 0), (0, 1), (1, 1), (0, 2)])
+        assert game.current_player() == X
+        game.change_player()
+        assert game.current_player() == O
+        game.change_player()
+        assert game.current_player() == X
+
+    def test_play_game_X_wins(self):
+        game = XOGame([(0, 0), (1, 0), (0, 1), (1, 1), (0, 2)])
+        game.play_game(5)
+        assert game.get_winner() == X
+
+    def test_play_game_O_wins(self):
+        game = XOGame([(0, 0), (1, 0), (0, 1), (1, 1), (2, 2), (1, 2)])
+        game.play_game(6)
+        assert game.get_winner() == O
+
+    def test_play_game_no_one_win(self):
+        game = XOGame([(1, 1), (0, 0), (2, 2), (2, 1), (0, 2)])
+        game.play_game(5)
+        assert game.get_winner() is None
+        game = XOGame([(1, 1), (1, 0), (2, 2), (2, 1), (0, 0)])
+        game.play_game(5)
+        assert game.get_winner() == X
+
+    def test_put_in_place_is_not_move_legal(self):
+        game = XOGame([(0, 0), (0, 0), (2, 2), (2, 1)])
+        now_player = game.current_player()
+        game.play_game(3)
+        assert game.current_player() != now_player
+
+    def test_check_is_vs_equality(self):
+        lst1 = []
+        lst2 = []
+        lst3 = lst1
+        assert lst1 is not lst2
+        assert lst1 == lst2
+        assert lst1 is lst3
+
+    def test_break_win(self):
+        gen = iter([(1, 1), (0, 1), (0, 0), (2, 0), (2, 2), (1, 0)])
+        game = XOGame(gen)
+        game.play_game(6)
+        assert next(gen) == (1, 0)
+
+    def test_place_not_legal_then_still_play_n_move_turns(self):
+        game = XOGame([(0, 0), (0, 0), (0, 0), (2, 0), (1, 1), (2, 1), (2, 2)])
+        start_player = game.current_player()
+        game.play_game(5)
+        assert game.get_winner() == start_player
+
+    def test_vs_comp_bool_value(self):
+        game = XOGame([(1, 1)], True)
+        start_player = game.current_player()
+        game.play_game(2)
+        end_player = game.current_player()
+        assert start_player == end_player
+
+        game = XOGame([(1, 1)], False)
+        start_player = game.current_player()
+        game.play_game(1)
+        end_player = game.current_player()
+        assert start_player != end_player
```

Comment:
Notice that these lines are independent of the lines above.
This should be a separate test case.